rust/lance-encoding/src/encodings/logical/primitive.rs

```diff
@@ -1101,17 +1131,78 @@ struct MiniBlockSchedulerDictionary {
     dictionary_data_alignment: u64,
 }
 
+#[derive(Debug)]
+struct RepIndexBlock {
+    // The first row in the block, if there is a preamble then this is the offset
+    // of the row after the preamble (that row may not exist if the block is entirely)
+    // preamble
```

Comment:
The paren above this line seems misplaced

Reply:
I rewrote this whole comment based on the discussion below:

```
// The index of the first row that starts after the beginning of this block.  If the block
// has a preamble this will be the row after the preamble.  If the block is entirely preamble
// then this will be a row that starts in some future block.
```